CORE-11431 - Added CpiMetadataRepository

filipesoliveira opened the pull request:
Queries related to the CpiMetadataEntity should always be done using the CpiMetadataRepository.
CpiMetadataEntity, CpiCpkMetadataEntitty, and CpkMetadataEntity are now internal classes.
Updated the tests related to CpiMetadataEntity, CpiCpkMetadataEntity, and CpkMetadataEntity.

filipesoliveira committed:
CORE-11431 - Added CpiMetadataRepository

Queries related to the CpiMetadataEntity should always be done using the CpiMetadataRepository.

filipesoliveira commented:
BEAR IN MIND THAT THE CUT-OFF IS DUE ON THE 5TH OF MAY. THIS PR CAN WAIT IF IT IS DEEMED TOO RISKY!

kyriathar commented on components/chunking/chunk-db-write-impl/src/integrationTest/kotlin/net/corda/chunking/db/impl/tests/UpsertCpiTests.kt:
```diff
@@ -1,6 +1,12 @@
 package net.corda.chunking.db.impl.tests
 
 import com.google.common.jimfs.Jimfs
+import java.nio.file.FileSystem
+import java.nio.file.Files
+import java.nio.file.Path
+import java.time.Instant
+import java.util.Random
+import java.util.UUID
```
I think these need to be put back at the bottom, your intelliJ might need some tweaking to change this behavior.

vkolomeyko commented on libs/virtual-node/cpi-datamodel/src/main/java/net/corda/libs/cpi/datamodel/repository/impl/package-info.java:
```diff
@@ -0,0 +1,4 @@
+@Export
+package net.corda.libs.cpi.datamodel.repository.impl;
```
I am somewhat concerned that we are exporting `impl` package from this module.

I would much rather keep `impl` classes as internal to this module and only export interfaces from package `net.corda.libs.cpi.datamodel.repository`.

When we do need to create concrete implementations, we might be better off introducing factories. This will avoid leaking `Impl` classes to other modules.

filipesoliveira replied:
I have applied the suggested change.

kyriathar commented on components/chunking/chunk-db-write-impl/src/main/kotlin/net/corda/chunking/db/impl/persistence/database/DatabaseCpiPersistence.kt:
```diff
@@ -1,40 +1,40 @@
-    private val networkInfoWriter: NetworkInfoWriter
+    private val networkInfoWriter: NetworkInfoWriter,
+    private val cpiMetadataRepository: CpiMetadataRepository,
+    private val groupPolicyParser: GroupPolicyParser.Companion
```
Should we also make properties added to constructor also the below repositories defined in the companion object so they could be mocked for testing purposes if needed?

filipesoliveira replied:
I see your point. Yeah, that is probably better. I will do that in my next PR, though.

filipesoliveira committed:
Added a factory for the cpi and cpk repositories.
The implementation of the cpi and cpk repositories are now internal and not exposed by OSGI.

filipesoliveira committed:
Fixed build issue.

filipesoliveira committed:
Fixed build issue.

filipesoliveira committed:
Fixed build issue.

filipesoliveira committed:
Fixed build issue.

vkolomeyko commented:
Looks like Detekt is not happy about:
```
/var/jenkins/workspace/Corda5_corda-runtime-os_PR-3696/components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterFactory.kt:84:13: Line detected, which is longer than the defined maximum line length in the code style. [MaxLineLength]	
/var/jenkins/workspace/Corda5_corda-runtime-os_PR-3696/components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterFactory.kt:162:13: Line detected, which is longer than the defined maximum line length in the code style. [MaxLineLength]	
```

filipesoliveira committed:
Fixed issue reported by detekt.

filipesoliveira committed:
Fixed tests.

vkolomeyko reviewed (approved):
LGTM

ronanbrowne reviewed (approved):
LGTM - build

kyriathar commented on components/chunking/chunk-db-write-impl/src/integrationTest/kotlin/net/corda/chunking/db/impl/tests/Helpers.kt:
```diff
@@ -1,15 +1,15 @@
 package net.corda.chunking.db.impl.tests
 
+import java.util.Random
+import java.util.UUID
```
These imports need to be at the bottom

kyriathar reviewed (approved):
Nice one. Some minor comments most to do with `java*` imports in wrong places I believe

filipesoliveira commented:
The remaining suggestions will be addressed in the following PR.

filipesoliveira commented:
PR to address the outstanding suggestions:
https://github.com/corda/corda-runtime-os/pull/4247/files
